konst a1: Int = <!INITIALIZER_TYPE_MISMATCH!>1L<!>
konst a2: Int = <!INITIALIZER_TYPE_MISMATCH!>0x1L<!>
konst a3: Int = <!INITIALIZER_TYPE_MISMATCH!>0X1L<!>
konst a4: Int = <!INITIALIZER_TYPE_MISMATCH!>0b1L<!>
konst a5: Int = <!INITIALIZER_TYPE_MISMATCH!>0B1L<!>
konst a6: Long = 1L
konst a7: Long = 0x1L
konst a8: Long = 0X1L
konst a9: Long = 0b1L
konst a10: Long = 0B1L