// FIR_IDENTICAL
konst a1: Int = 1<!WRONG_LONG_SUFFIX!>l<!>
konst a2: Int = 0x1<!WRONG_LONG_SUFFIX!>l<!>
konst a3: Int = 0X1<!WRONG_LONG_SUFFIX!>l<!>
konst a4: Int = 0b1<!WRONG_LONG_SUFFIX!>l<!>
konst a5: Int = 0B1<!WRONG_LONG_SUFFIX!>l<!>
konst a6: Long = 1<!WRONG_LONG_SUFFIX!>l<!>
konst a7: Long = 0x1<!WRONG_LONG_SUFFIX!>l<!>
konst a8: Long = 0X1<!WRONG_LONG_SUFFIX!>l<!>
konst a9: Long = 0b1<!WRONG_LONG_SUFFIX!>l<!>
konst a10: Long = 0B1<!WRONG_LONG_SUFFIX!>l<!>