// FIR_IDENTICAL
class <!DANGEROUS_CHARACTERS!>`A?B`<!>
class <!DANGEROUS_CHARACTERS!>`A*B`<!>
class <!DANGEROUS_CHARACTERS!>`A"B`<!>
class <!DANGEROUS_CHARACTERS!>`A|B`<!>
class <!DANGEROUS_CHARACTERS!>`A%B`<!>

fun <!DANGEROUS_CHARACTERS!>`?*"|%`<!>(): Int {
    konst <!DANGEROUS_CHARACTERS!>`?`<!> = 0
    return `?`
}

konst <!DANGEROUS_CHARACTERS!>`"a"+"b"`<!> = "c"
