/*
 * KOTLIN DIAGNOSTICS SPEC TEST (POSITIVE)
 *
 * SPEC VERSION: 0.1-435
 * MAIN LINK: expressions, constant-literals, real-literals -> paragraph 2 -> sentence 1
 * NUMBER: 1
 * DESCRIPTION: Simple real literals with an exponent mark.
 */

// TESTCASE NUMBER: 1
konst konstue_1 = 0.0e0

// TESTCASE NUMBER: 2
konst konstue_2 = 0.0e-00

// TESTCASE NUMBER: 3
konst konstue_3 = 0.0E000

// TESTCASE NUMBER: 4
konst konstue_4 = 0.0E+0000

// TESTCASE NUMBER: 5
konst konstue_5 = 00.0e+0

// TESTCASE NUMBER: 6
konst konstue_6 = 000.00e00

// TESTCASE NUMBER: 7
konst konstue_7 = 0000.000E-000

// TESTCASE NUMBER: 8
konst konstue_8 = 1.0E+1

// TESTCASE NUMBER: 9
konst konstue_9 = 22.00e22

// TESTCASE NUMBER: 10
konst konstue_10 = <!FLOAT_LITERAL_CONFORMS_ZERO!>333.000e-333<!>

// TESTCASE NUMBER: 11
konst konstue_11 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>4444.0000e4444<!>

// TESTCASE NUMBER: 12
konst konstue_12 = <!FLOAT_LITERAL_CONFORMS_ZERO!>55555.0E-55555<!>

// TESTCASE NUMBER: 13
konst konstue_13 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>666666.00e666666<!>

// TESTCASE NUMBER: 14
konst konstue_14 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>7777777.000E7777777<!>

// TESTCASE NUMBER: 15
konst konstue_15 = <!FLOAT_LITERAL_CONFORMS_ZERO!>88888888.0000e-88888888<!>

// TESTCASE NUMBER: 16
konst konstue_16 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>999999999.0E+999999999<!>

// TESTCASE NUMBER: 17
konst konstue_17 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>0000000000.1234567890E999999999<!>

// TESTCASE NUMBER: 18
konst konstue_18 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>123456789.23456789E+123456789<!>

// TESTCASE NUMBER: 19
konst konstue_19 = 2345678.345678e00000000001

// TESTCASE NUMBER: 20
konst konstue_20 = <!FLOAT_LITERAL_CONFORMS_ZERO!>34567.4567E-100000000000<!>

// TESTCASE NUMBER: 21
konst konstue_21 = 456.56e-0

// TESTCASE NUMBER: 22
konst konstue_22 = 5.65e000000000000

// TESTCASE NUMBER: 23
konst konstue_23 = 654.7654E+010

// TESTCASE NUMBER: 24
konst konstue_24 = 76543.876543E1

// TESTCASE NUMBER: 25
konst konstue_25 = 8765432.98765432e-2

// TESTCASE NUMBER: 26
konst konstue_26 = 987654321.0987654321E-3

// TESTCASE NUMBER: 27
konst konstue_27 = 0.1111e4

// TESTCASE NUMBER: 28
konst konstue_28 = 1.22222E-5

// TESTCASE NUMBER: 29
konst konstue_29 = 9.33333e+6

// TESTCASE NUMBER: 30
konst konstue_30 = 9.444444E7

// TESTCASE NUMBER: 31
konst konstue_31 = 8.5555555e8

// TESTCASE NUMBER: 32
konst konstue_32 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>2.66666666e308<!>

// TESTCASE NUMBER: 33
konst konstue_33 = 3.777777777E-308

// TESTCASE NUMBER: 34
konst konstue_34 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>7.8888888888e+309<!>

// TESTCASE NUMBER: 35
konst konstue_35 = 6.99999999999e-309

// TESTCASE NUMBER: 36
konst konstue_36 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>7.888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888e+111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111<!>

// TESTCASE NUMBER: 37
konst konstue_37 = 0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000

// TESTCASE NUMBER: 38
konst konstue_38 = 0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e-000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000

// TESTCASE NUMBER: 39
konst konstue_39 = 0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e+000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
