// TESTCASE NUMBER: 1
konst konstue_1 = <!ILLEGAL_UNDERSCORE!>0b0_l<!>

// TESTCASE NUMBER: 2
konst konstue_2 = <!ILLEGAL_UNDERSCORE!>0B12_L<!>

// TESTCASE NUMBER: 3
konst konstue_3 = <!ILLEGAL_UNDERSCORE!>0X234_l<!>

// TESTCASE NUMBER: 4
konst konstue_4 = <!ILLEGAL_UNDERSCORE!>0x3567_L<!>
