// FIR_IDENTICAL

fun main() {
    konst appendable = <!NO_COMPANION_OBJECT!>Appendable<!>
    konst stringBuilder = <!NO_COMPANION_OBJECT!>StringBuilder<!>
    konst exception = <!NO_COMPANION_OBJECT!>IllegalStateException<!>
}
