// FIR_IDENTICAL
class `___` {
    class `____`
}

konst testCallableRefLHSType = <!UNDERSCORE_USAGE_WITHOUT_BACKTICKS!>___<!>::toString
konst testCallableRefLHSType2 = `___`::toString

konst testClassLiteralLHSType = <!UNDERSCORE_USAGE_WITHOUT_BACKTICKS!>___<!>::class
konst testClassLiteralLHSType2 = `___`::class

konst tesLHSTypeFQN = `___`.<!UNDERSCORE_USAGE_WITHOUT_BACKTICKS!>____<!>::class