// TESTCASE NUMBER: 1
konst konstue_1 = <!UNRESOLVED_REFERENCE!>E0f<!>

// TESTCASE NUMBER: 2
konst konstue_2 = <!UNRESOLVED_REFERENCE!>e000F<!>

// TESTCASE NUMBER: 3
konst konstue_3 = <!UNRESOLVED_REFERENCE!>E<!>+0f

// TESTCASE NUMBER: 4
konst konstue_4 = <!UNRESOLVED_REFERENCE!>e00f<!>

// TESTCASE NUMBER: 5
konst konstue_5 = <!UNRESOLVED_REFERENCE!>e<!>+1F

// TESTCASE NUMBER: 6
konst konstue_6 = <!UNRESOLVED_REFERENCE!>e22F<!>

// TESTCASE NUMBER: 7
konst konstue_7 = <!UNRESOLVED_REFERENCE!>E<!>-333F

// TESTCASE NUMBER: 8
konst konstue_8 = <!UNRESOLVED_REFERENCE!>e4444f<!>

// TESTCASE NUMBER: 9
konst konstue_9 = <!UNRESOLVED_REFERENCE!>e<!>-55555f

// TESTCASE NUMBER: 10
konst konstue_10 = <!UNRESOLVED_REFERENCE!>e666666F<!>

// TESTCASE NUMBER: 11
konst konstue_11 = <!UNRESOLVED_REFERENCE!>E7777777f<!>

// TESTCASE NUMBER: 12
konst konstue_12 = <!UNRESOLVED_REFERENCE!>e<!>-88888888F

// TESTCASE NUMBER: 13
konst konstue_13 = <!UNRESOLVED_REFERENCE!>E<!>+999999999F
