konst <!ELEMENT!> = 10
konst <!ELEMENT!> = "."

konst konstue_1 = <!ELEMENT!> - 100 % <!ELEMENT!>
konst konstue_2 = <!ELEMENT!>.dec()
konst konstue_3 = "$<!ELEMENT!> 10"
konst konstue_4 = "${<!ELEMENT!>}"
konst konstue_5 = <!ELEMENT!> + " 11..." + <!ELEMENT!> + "1"
konst konstue_6 = <!ELEMENT!>
