fun f1(<!ELEMENT!>: Int) {}

fun f2(<!ELEMENT!>: Boolean, <!ELEMENT!>: Boolean) = <!ELEMENT!>

fun f3(konst <!ELEMENT!>: Boolean = <!ELEMENT!>, var <!ELEMENT!>: Boolean) = <!ELEMENT!>

class A {
    konst x1: Boolean = <!ELEMENT!>
        set(<!ELEMENT!>) = <!ELEMENT!>
    var x2: Boolean = <!ELEMENT!>
        set(vararg <!ELEMENT!>: Boolean) = <!ELEMENT!>
}
