// TESTCASE NUMBER: 1
konst konstue_1= <!ILLEGAL_UNDERSCORE!>.0_0e-0___<!>

// TESTCASE NUMBER: 2
konst konstue_2 = <!ILLEGAL_UNDERSCORE!>0_0___0E+10_<!>

// TESTCASE NUMBER: 3
konst konstue_3 = <!ILLEGAL_UNDERSCORE!>.0_0e-0_F<!>

// TESTCASE NUMBER: 4
konst konstue_4 = <!ILLEGAL_UNDERSCORE!>0__________________________0E+3___<!>

// TESTCASE NUMBER: 5
konst konstue_5 = <!ILLEGAL_UNDERSCORE!>0_00e1_<!>

// TESTCASE NUMBER: 6
konst konstue_6 = <!ILLEGAL_UNDERSCORE!>.0_0__0___0E0_______________<!>

// TESTCASE NUMBER: 7
konst konstue_7 = <!ILLEGAL_UNDERSCORE!>0_1e+100_____f<!>

// TESTCASE NUMBER: 8
konst konstue_8 = <!ILLEGAL_UNDERSCORE!>.0_0E-0001___<!>

// TESTCASE NUMBER: 9
konst konstue_9 = <!ILLEGAL_UNDERSCORE!>0___0__0e4___<!>

// TESTCASE NUMBER: 10
konst konstue_10 = <!ILLEGAL_UNDERSCORE!>0_00_0e99_f<!>

// TESTCASE NUMBER: 11
konst konstue_11 = <!ILLEGAL_UNDERSCORE!>.0_5E-93_______F<!>

// TESTCASE NUMBER: 12
konst konstue_12 = <!ILLEGAL_UNDERSCORE!>.8____8888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888_____8888888888888888888e+5_f<!>

// TESTCASE NUMBER: 13
konst konstue_13 = <!ILLEGAL_UNDERSCORE!>0_0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000_0e40_F<!>

// TESTCASE NUMBER: 14
konst konstue_14 = <!ILLEGAL_UNDERSCORE!>0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_0_e-10_f<!>

// TESTCASE NUMBER: 15
konst konstue_15 = <!ILLEGAL_UNDERSCORE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e+75_<!>
