// TESTCASE NUMBER: 1
konst konstue_1 = <!ILLEGAL_UNDERSCORE!>0_E0<!>

// TESTCASE NUMBER: 2
konst konstue_2 = <!ILLEGAL_UNDERSCORE!>1.1_E2<!>

// TESTCASE NUMBER: 3
konst konstue_3 = <!ILLEGAL_UNDERSCORE!>.0__0_e-0___0<!>

// TESTCASE NUMBER: 4
konst konstue_4 = <!ILLEGAL_UNDERSCORE!>0__0.0_____e0f<!>

// TESTCASE NUMBER: 5
konst konstue_5 = <!ILLEGAL_UNDERSCORE!>9_______9______9_____9____9___9__9_9.0___E-1<!>

// TESTCASE NUMBER: 6
konst konstue_6 = <!ILLEGAL_UNDERSCORE!>0_0_0_0_0_0_0_0_0_0.12345678___e+90F<!>

// TESTCASE NUMBER: 7
konst konstue_7 = <!ILLEGAL_UNDERSCORE!>1_2_3_4_5_6_7_8_9.2_3_4_5_6_7_8_9_e-0<!>

// TESTCASE NUMBER: 8
konst konstue_8 = <!ILLEGAL_UNDERSCORE!>456.5__e0_6<!>

// TESTCASE NUMBER: 9
konst konstue_9 = <!ILLEGAL_UNDERSCORE!>5.6_0_E+05F<!>

// TESTCASE NUMBER: 10
konst konstue_10 = <!ILLEGAL_UNDERSCORE!>6_54.76_5__e-4<!>

// TESTCASE NUMBER: 11
konst konstue_11 = <!ILLEGAL_UNDERSCORE!>9_____________87654321.0__e-9_8765432_____________1F<!>

// TESTCASE NUMBER: 12
konst konstue_12 = <!ILLEGAL_UNDERSCORE!>000000000000000000000000000000000000000000000000000000000000000000000000000000000000000___0.000000000000000000000000_e000000000000000000000000000000000000000000000000000000000000000_0F<!>

// TESTCASE NUMBER: 13
konst konstue_13 = <!ILLEGAL_UNDERSCORE!>0_000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.0____E-0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000<!>

// TESTCASE NUMBER: 14
konst konstue_14 = <!ILLEGAL_UNDERSCORE!>9999999999999999999999999999999999999999999_______________999999999999999999999999999999999999999999999.33333333333333333333333333333333333333333333333_333333333333333333333333333333333333333_e3_3f<!>
