// FIR_IDENTICAL
/*
 * KOTLIN DIAGNOSTICS SPEC TEST (NEGATIVE)
 *
 * SPEC VERSION: 0.1-100
 * MAIN LINK: expressions, constant-literals, real-literals -> paragraph 3 -> sentence 1
 * NUMBER: 1
 * DESCRIPTION: Real literals with omitted a whole-number part and an exponent mark without digits after it.
 */

// TESTCASE NUMBER: 1
konst konstue_1 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0E<!>

// TESTCASE NUMBER: 2
konst konstue_2 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0e<!>

// TESTCASE NUMBER: 3
konst konstue_3 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.00e-<!>

// TESTCASE NUMBER: 4
konst konstue_4 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000E+<!>

// TESTCASE NUMBER: 5
konst konstue_5 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0e<!>

// TESTCASE NUMBER: 6
konst konstue_6 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.00E+<!>

// TESTCASE NUMBER: 7
konst konstue_7 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000e<!>

// TESTCASE NUMBER: 8
konst konstue_8 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0000E<!>

// TESTCASE NUMBER: 9
konst konstue_9 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0e+<!>

// TESTCASE NUMBER: 10
konst konstue_10 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.00E-<!>

// TESTCASE NUMBER: 11
konst konstue_11 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000e<!>

// TESTCASE NUMBER: 12
konst konstue_12 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0000e<!>

// TESTCASE NUMBER: 13
konst konstue_13 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0E-<!>

// TESTCASE NUMBER: 14
konst konstue_14 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888e+<!>

// TESTCASE NUMBER: 15
konst konstue_15 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e<!>

// TESTCASE NUMBER: 16
konst konstue_16 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e-<!>

// TESTCASE NUMBER: 17
konst konstue_17 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e+<!>
