// FIR_IDENTICAL
/*
 * KOTLIN DIAGNOSTICS SPEC TEST (NEGATIVE)
 *
 * SPEC VERSION: 0.1-100
 * MAIN LINK: expressions, constant-literals, real-literals -> paragraph 3 -> sentence 1
 * NUMBER: 2
 * DESCRIPTION: Real literals with omitted a whole-number part, suffixed by f/F (float suffix) followed by an exponent mark without digits.
 */

// TESTCASE NUMBER: 1
konst konstue_1 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0EF<!>

// TESTCASE NUMBER: 2
konst konstue_2 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0ef<!>

// TESTCASE NUMBER: 3
konst konstue_3 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.00e-f<!>

// TESTCASE NUMBER: 4
konst konstue_4 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000E+F<!>

// TESTCASE NUMBER: 5
konst konstue_5 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0eF<!>

// TESTCASE NUMBER: 6
konst konstue_6 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.00E+f<!>

// TESTCASE NUMBER: 7
konst konstue_7 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000ef<!>

// TESTCASE NUMBER: 8
konst konstue_8 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0000Ef<!>

// TESTCASE NUMBER: 9
konst konstue_9 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0e+F<!>

// TESTCASE NUMBER: 10
konst konstue_10 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.00E-F<!>

// TESTCASE NUMBER: 11
konst konstue_11 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000eF<!>

// TESTCASE NUMBER: 12
konst konstue_12 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0000ef<!>

// TESTCASE NUMBER: 13
konst konstue_13 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.0E-F<!>

// TESTCASE NUMBER: 14
konst konstue_14 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888e+f<!>

// TESTCASE NUMBER: 15
konst konstue_15 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000eF<!>

// TESTCASE NUMBER: 16
konst konstue_16 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e-f<!>

// TESTCASE NUMBER: 17
konst konstue_17 = <!FLOAT_LITERAL_OUT_OF_RANGE!>.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e+F<!>
