interface A<!CONSTRUCTOR_IN_INTERFACE!>(konst s: String)<!>

interface B <!CONSTRUCTOR_IN_INTERFACE!>constructor(konst s: String)<!>

interface C {
    <!CONSTRUCTOR_IN_INTERFACE!>constructor(s: String)<!> {}
}
