/*
 * KOTLIN DIAGNOSTICS SPEC TEST (POSITIVE)
 *
 * SPEC VERSION: 0.1-100
 * MAIN LINK: expressions, constant-literals, real-literals -> paragraph 3 -> sentence 2
 * NUMBER: 1
 * DESCRIPTION: Simple real literals with omitted a fraction part, suffixed by f/F (float suffix).
 */

// TESTCASE NUMBER: 1
konst konstue_1 = 0F

// TESTCASE NUMBER: 2
konst konstue_2 = 00F

// TESTCASE NUMBER: 3
konst konstue_3 = 000f

// TESTCASE NUMBER: 4
konst konstue_4 = 0000F

// TESTCASE NUMBER: 5
konst konstue_5 = 00000000000000000000000000000000000000f

// TESTCASE NUMBER: 6
konst konstue_6 = 1f

// TESTCASE NUMBER: 7
konst konstue_7 = 22f

// TESTCASE NUMBER: 8
konst konstue_8 = 333F

// TESTCASE NUMBER: 9
konst konstue_9 = 4444f

// TESTCASE NUMBER: 10
konst konstue_10 = 55555F

// TESTCASE NUMBER: 11
konst konstue_11 = 666666f

// TESTCASE NUMBER: 12
konst konstue_12 = 7777777f

// TESTCASE NUMBER: 13
konst konstue_13 = 88888888F

// TESTCASE NUMBER: 14
konst konstue_14 = 999999999F

// TESTCASE NUMBER: 15
konst konstue_15 = 123456789f

// TESTCASE NUMBER: 16
konst konstue_16 = 2345678F

// TESTCASE NUMBER: 17
konst konstue_17 = 34567F

// TESTCASE NUMBER: 18
konst konstue_18 = 456F

// TESTCASE NUMBER: 19
konst konstue_19 = 5f

// TESTCASE NUMBER: 20
konst konstue_20 = 654F

// TESTCASE NUMBER: 21
konst konstue_21 = 76543F

// TESTCASE NUMBER: 22
konst konstue_22 = 8765432f

// TESTCASE NUMBER: 23
konst konstue_23 = 987654321F

// TESTCASE NUMBER: 24
konst konstue_24 = <!FLOAT_LITERAL_CONFORMS_INFINITY!>888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888888111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111F<!>

// TESTCASE NUMBER: 25
konst konstue_25 = <!FLOAT_LITERAL_CONFORMS_ZERO!>0.000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001f<!>
