konst <!REDECLARATION!>Test1<!> = null
class <!REDECLARATION!>Test1<!>

konst <!REDECLARATION!>Test2<!> = null
interface <!REDECLARATION!>Test2<!>

konst <!REDECLARATION!>Test3<!> = null
object <!REDECLARATION!>Test3<!>

konst <!REDECLARATION!>Test4<!> = null
class <!PACKAGE_OR_CLASSIFIER_REDECLARATION!>Test4<!>
interface <!PACKAGE_OR_CLASSIFIER_REDECLARATION!>Test4<!>
object <!PACKAGE_OR_CLASSIFIER_REDECLARATION!>Test4<!>